### PR TITLE
feat: adding new issue templates format + `bentoml env`

### PR DESCRIPTION
**Pull request by aarnphm**
## What does this PR address?

update new PR templates and issue templates https://github.blog/changelog/2021-06-23-issues-forms-beta-for-public-repositories/

for context: 
Feature request templates
https://github.com/aarnphm/BentoML/blob/feat%2Fnew_issue_format/.github/ISSUE_TEMPLATE/feature_request.yml

Issue templates
https://github.com/aarnphm/BentoML/blob/feat/new_issue_format/.github/ISSUE_TEMPLATE/bug_report.yml

This PR also add an easier `bentoml env` command to pretty print to markdown

## Before submitting:

- [x] Does the Pull Request follow [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary) naming?
- [x] Does the code follow BentoML's code style, both `make format` and `make lint` script have passed ([instructions](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#style-check-auto-formatting-type-checking))?
- [x] Did you read through [contribution guidelines](https://github.com/bentoml/BentoML/blob/main/CONTRIBUTING.md#ways-to-contribute) and follow [development guidelines](https://github.com/bentoml/BentoML/blob/main/DEVELOPMENT.md#start-developing)?
- [x] Did your changes require updates to the documentation? Have you updated
  those accordingly? Here are [documentation guidelines](https://github.com/bentoml/BentoML/tree/main/docs) and [tips on writting docs](https://github.com/bentoml/BentoML/tree/main/docs#writing-documentation).
- [x] Did you write tests to cover your changes?

## Who can help review?

cc @ssheng @parano

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  - type: checkboxes
+    id: issue-already-exists
+    attributes:
+      label: Is there an existing issue for this?
+      description: |
+        Please search to see if an issue already exists for the bug you encountered.
+        Please see [Searching Issues and Pull Request](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for how to use the GitHub search bar and filters.
+      options:
+        label: I have searched the existing issues
+        required: true
```
Comment (ssheng) [inline .github/ISSUE_TEMPLATE/bug_report.yml]: Github reported an error here.
```
body[0]: options must be of type Array.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+title: "bug: "
+labels: ["bug"]
+body:
+  - type: checkboxes
+    id: issue-already-exists
+    attributes:
+      label: Is there an existing issue for this?
+      description: |
+        Please search to see if an issue already exists for the bug you encountered.
+        Please see [Searching Issues and Pull Requests](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for how to use the GitHub search bar and filters.
```
Comment (sauyon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:12]:
```suggestion
        See [Searching Issues and Pull Requests](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for how to use the GitHub search bar and filters.
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  - type: textarea
+    id: environment-info
+    attributes:
+      label: Environment
+      description: |
+        Please share your environment with us. You should run `bentoml env` and paste the result here.
+      placeholder: |
+        bentoml: 1.0.0rc3
+        python: 3.10.5
+        platform: macOS-12.4-arm64-arm-64bit
```
Comment (sauyon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:76]:
```suggestion
        bentoml: ...
        python: ...
        platform: ...
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
     about: Please ask general questions here.
   - name: BentoML Slack Community
-    url: https://join.slack.bentoml.org
-    about: For connecting with others in the community
+    url: https://l.linklyhq.com/l/ktOh
+    about: For connecting with others in the community.
   - name: BentoML Enterprise Support
     url: contact@bentoml.ai
-    about: Contact the BentoML team and learn more
+    about: Contact the BentoML team for supports.
```
Comment (sauyon) [inline .github/ISSUE_TEMPLATE/config.yml:18]:
```suggestion
    about: Contact the BentoML team for support.
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 5]
```diff
@@ -1,10 +1,21 @@
+blank_issues_enabled: true
+version: 2.1
 contact_links:
+  - name: BentoML's Issue Tracker
+    url: https://github.com/bentoml/BentoML/issues/new/choose
+    about: Open a PR/Issue ticket for BentoML's development.
```
Comment (sauyon) [inline .github/ISSUE_TEMPLATE/config.yml:6]:
```suggestion
```

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+- type: textarea
+  id: feature-request
+  validations:
+    required: true
+  attributes:
+    label: Feature request
+    description: |
+      A clear and concise description of the feature request.
+    placeholder: |
+      An implementation could be implemented by...
```
Comment (sauyon) [inline .github/ISSUE_TEMPLATE/feature_request.yml:15]:
```suggestion
      I would like it if...
```

File: bentoml/_internal/cli/__init__.py [hunk 7]
```diff
@@ -22,29 +21,79 @@
...
-        if not value or ctx.resilient_parsing:
-            return
+def format_dropdown(title: str, content: t.Iterable[str]) -> str:
+    processed = "\n".join(content)
+    return f"""\
+<details><summary><code>{title}</code></summary>
 
+<br>
+
+```{ 'yaml' if title == conda_packages_name else 'markdown' }
```
Comment (sauyon) [inline bentoml/_internal/cli/__init__.py:38]:
```suggestion
```{ 'yaml' if title == conda_packages_name else '' }
```

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    attributes:
+      label: Is there an existing issue for this?
+      description: |
+        Please search to see if an issue already exists for the bug you encountered.
+        Please see [Searching Issues and Pull Requests](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests) for how to use the GitHub search bar and filters.
+      options:
+      - label: I have searched the existing issues
+        required: true
+  - type: textarea
+    id: describe-the-bugs
```
Comment (aarnphm) [inline .github/ISSUE_TEMPLATE/bug_report.yml:17]:
```suggestion
    id: describe-the-bug
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 9]
```diff
@@ -1,10 +1,18 @@
...
+  - name: BentoML's Documentation
+    url: https://docs.bentoml.org/en/latest/
+    about: For more information about BentoML's tutorials, guides, API references and architectures deep-dive.
   - name: BentoML Discussions
     url: https://github.com/bentoml/BentoML/discussions
     about: Please ask general questions here.
   - name: BentoML Slack Community
-    url: https://join.slack.bentoml.org
-    about: For connecting with others in the community
+    url: https://l.linklyhq.com/l/ktOh
```
Comment (ssheng) [inline .github/ISSUE_TEMPLATE/config.yml:11]:
```suggestion
    url: https://join.slack.bentoml.org
```
```suggestion
    url: https://l.linklyhq.com/l/ktOh
```

**Review by ssheng — APPROVED** [review]

**Review by ssheng — APPROVED** [review]

**Pull request merged**